### PR TITLE
Add 'forceLocalizationDefaults' config option

### PR DESCRIPTION
**Pull request by DanFerrer**
### This PR will...

- Add new config option ```forceLocalizationDefaults```
- Add unit test for config option:
  -  Check that page language doesn't change.
  -  Check that we don't try to load translation files if config option is ```true```
  -  Check we don't use ```intl``` block if given.
  -  Check localization options in config block still work.

### Why is this Pull Request needed?

New a/b test

### Are there any Pull Requests open in other repos which need to be merged with this?

https://github.com/jwplayer/borg/pull/1056
https://github.com/jwplayer/jwplayer-commercial/pull/5859

#### Addresses Issue(s):

JW8-2260

### CODE REVIEW

File: test/unit/language-test.js [hunk 1]
```diff
@@ -337,6 +341,30 @@ describe('languageUtils', function() {
...
+                intl
+            });
+            const configLocalization = config.localization;
+            expect(getLanguage()).to.equal('fr');
+            expect(configLocalization.play).to.equal(localizationPlay);
+            expect(configLocalization.pause).to.equal(localizationPause);
+            expect(configLocalization.stop).to.equal(localizationStop);
+            
+            Object.keys(en).forEach(key => {
+                if (key === 'play' || key === 'pause' || key === 'stop') {
```
Comment (karimJWP) [inline test/unit/language-test.js]: we don't need this if block right ? Since what we care is that all localization values match the default values.
I assume you had this because those values are the same in our french translation, but that doesn't matter.

File: test/unit/language-test.js [hunk 2]
```diff
@@ -337,6 +341,30 @@ describe('languageUtils', function() {
...
+        it('should only use custom localization block if "forceLocalizationDefaults" is true', function() {
+            stubHtmlLanguage(document, 'fr');
+            const config = new Config({
+                forceLocalizationDefaults: true,
+                localization,
+                intl
+            });
+            const configLocalization = config.localization;
+            expect(getLanguage()).to.equal('fr');
+            expect(configLocalization.play).to.equal(localizationPlay);
```
Comment (karimJWP) [inline test/unit/language-test.js]: I would add a check that config.language === 'en'

File: test/unit/language-test.js [hunk 3]
```diff
@@ -337,6 +341,30 @@ describe('languageUtils', function() {
             expect(customLocalization.pause).to.equal(localizationPause);
             expect(customLocalization.stop).to.equal(localizationStop);
         });
+
+        it('should only use custom localization block if "forceLocalizationDefaults" is true', function() {
+            stubHtmlLanguage(document, 'fr');
```
Comment (karimJWP) [inline test/unit/language-test.js]: No need to stub the language on the hmtl, we can stub `getLanguage` to return 'fr'
import the entire language module `import * as Language from 'utils/language';`
`sandbox.stub(Language, 'getLanguage').returns('fr');`

File: test/unit/language-test.js [hunk 4]
```diff
@@ -309,6 +309,10 @@ describe('languageUtils', function() {
             };
         });
 
+        after(function () {
```
Comment (karimJWP) [inline test/unit/language-test.js]: I would move your unit test to it's own `describe` block. Make one for `forceLocalizationDefaults`

File: test/unit/language-test.js [hunk 5]
```diff
@@ -337,6 +351,21 @@ describe('languageUtils', function() {
...
+
+        it('should only use custom localization block if "forceLocalizationDefaults" is true', function() {
+            sandbox.stub(Language, 'getLanguage').returns('fr');
+            const config = new Config({
+                forceLocalizationDefaults: true,
+                localization,
+                intl
+            });
+            const configLocalization = config.localization;
+            expect(Language.getLanguage()).to.equal('fr');
```
Comment (karimJWP) [inline test/unit/language-test.js]: we don't need to test this; it would mean we are testing the stub

File: src/js/api/config.js [hunk 6]
```diff
@@ -119,7 +119,7 @@ const Config = function(options, persisted) {
     _copyToLocalization(allOptions);
     _deserialize(allOptions);
 
-    const language = getLanguage();
+    const language = allOptions.forceLocalizationDefaults ? Defaults.language : getLanguage();
```
Comment (robwalch) [inline src/js/api/config.js]: What is the value of `Defaults.language`? Why don't we just set a language in the config, and base the test on `setupConfig.language`? I much rather we use existing config options and support the setting of `language` in the config, than add a new option that no one need like `forceLocalizationDefaults`.

Comment (karimJWP) [reply src/js/api/config.js]: `Defaults.language` is 'en'. I have mixed feelings about providing a config option to override language because we want to support google translate, which overrides the language tag on the page. If the player is setup after google translate then we read the updated language tag.
 if the language is being overriden via the config by the developer, then we have to count on them to update it when the page has been translated, which i doubt they would do.

File: test/unit/language-test.js [hunk 7]
```diff
@@ -353,18 +343,23 @@ describe('languageUtils', function() {
         });
 
         it('should only use custom localization block if "forceLocalizationDefaults" is true', function() {
-            sandbox.stub(Language, 'getLanguage').returns('fr');
```
Comment (karimJWP) [inline test/unit/language-test.js]: @DanFerrer why did you remove the stubbing ? If we don't stub then how are we verifying that we override the language when `forceLocalizationDefaults` is true ?

Comment (karimJWP) [reply test/unit/language-test.js]: I think you might have misunderstood our conversation from friday. What i meant to say was that `expect(Language.getLanguage()).to.equal('fr');` is unnecessary since get language is being stubbed and we don't need to unit test the stubbing. But in order to properly test that `forceLocalizationDefaults ` overrides the language, we need to stub `getLanguage`, makes sense ?
